src/Analyzers/CSharp/CodeFixes/UseImplicitOrExplicitType/UseExplicitTypeCodeFixProvider.cs

```diff
@@ -63,63 +62,123 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
+                var typeSymbol = semanticModel.GetTypeInfo(typeSyntax, cancellationToken).ConvertedType;
+                RoslynDebug.AssertNotNull(typeSymbol);
+
+                editor.ReplaceNode(typeSyntax, GenerateTypeDeclaration(typeSyntax, typeSymbol));
+            }
+        }
+
+        private static async Task HandleForEachStatementAsync(Document document, SyntaxEditor editor, ForEachStatementSyntax forEach, CancellationToken cancellationToken)
+        {
+            var typeSyntax = forEach.Type.StripRefIfNeeded();
+            var declarationSyntax = forEach.Identifier.Parent;
+
+            var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var typeSymbol = semanticModel.GetTypeInfo(typeSyntax, cancellationToken).ConvertedType;
+
+            RoslynDebug.AssertNotNull(typeSymbol);
+            RoslynDebug.AssertNotNull(declarationSyntax);
+
+            typeSymbol = AdjustNullabilityOfTypeSymbol(
+                typeSymbol,
+                semanticModel,
+                declarationSyntax,
+                cancellationToken);
 
-                var leadingTrivia = node.GetLeadingTrivia()
-                    .Concat(parensDesignation.GetAllPrecedingTriviaToPreviousToken().Where(t => !t.IsWhitespace()).Select(t => t.WithoutAnnotations(SyntaxAnnotation.ElasticAnnotation)));
+            editor.ReplaceNode(typeSyntax, GenerateTypeDeclaration(typeSyntax, typeSymbol));
+        }
+
+        private static async Task HandleVariableDeclarationAsync(Document document, SyntaxEditor editor, VariableDeclarationSyntax varDecl, CancellationToken cancellationToken)
+        {
+            var typeSyntax = varDecl.Type.StripRefIfNeeded();
+            var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+            var typeSymbol = semanticModel.GetTypeInfo(typeSyntax, cancellationToken).ConvertedType;
+            RoslynDebug.AssertNotNull(typeSymbol);
+
+            // Since we're only dealing with variable declaration using var, we know
+            // that implicitly typed variables cannot have multiple declarators in
+            // a single declaration (CS0819). Only one variable should be present
+            var declarationSyntax = varDecl.Variables.Single().Identifier.Parent;
+            RoslynDebug.AssertNotNull(declarationSyntax);
 
-                var tupleDeclaration = GenerateTupleDeclaration(tupleTypeSymbol, parensDesignation).WithLeadingTrivia(leadingTrivia);
+            typeSymbol = AdjustNullabilityOfTypeSymbol(
+                typeSymbol,
+                semanticModel,
+                declarationSyntax,
+                cancellationToken);
 
-                editor.ReplaceNode(declarationContext, tupleDeclaration);
+            editor.ReplaceNode(typeSyntax, GenerateTypeDeclaration(typeSyntax, typeSymbol));
```

Comment:
this method and hte method above seem almost identical.  any way to share more?

Reply:
Sure